### PR TITLE
Chore: parse pdf as markdown and remove vendor

### PR DESCRIPTION
**Pull request by StrongMonkey**
This PR adds the following things:

1. Switch to using https://github.com/gen2brain/go-fitz instead of unidoc. Although the underlying mupdf lib is still AGPL, we can use the latest code as long as we stay in open source.

2. Parse pdf text as markdown and store it in vector DB. this seems to improve the accuracy in how LLM understand it.

3. Remove all the vendor files. It happens that when using https://github.com/gen2brain/go-fitz, we can't use vendor because it doesn't include c code by default. see https://github.com/gen2brain/go-fitz/issues/60 and https://github.com/golang/go/issues/27667.

With the change, we only have three test cases failing!

image

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -12,7 +12,7 @@ endif
 GO_TAGS := netgo
 LD_FLAGS := -s -w -X github.com/gptscript-ai/knowledge/version.Version=${GIT_TAG}
 build:
-	CGO_ENABLED=0 go build -o bin/knowledge -tags "${GO_TAGS}" -ldflags '$(LD_FLAGS)' .
+	go build -o bin/knowledge -tags "${GO_TAGS}" -ldflags '$(LD_FLAGS)' .
```
Comment (g-linville) [inline Makefile:15]: Are we certain that turning on CGo here doesn't cause any issues on each OS?

**Review by iwilltry42 — APPROVED** [review]
If this works across all OSes, then this is great!

File: Makefile (hunk 1 above)
Comment (StrongMonkey) [reply Makefile:15]: Yeah, tbh I haven't tested on other OS, but the c library is linked from go module. Will test on other os too 👍

**Comment by StrongMonkey** [comment]
I have tested this PR against linux and windows, both works fine. 

Although we need to modify the build process to do cross-compiling now as it involves C library. This is a bit tricky and I will have a PR later to fix the build and CI.

**Pull request merged**